Translate additionalProperties API properties into arrays.

bmenasha opened the pull request:
This changes the BigQuery schema and is not backwards compatible with prior
imports. Subsequent WRITE_APPEND import runs will fail with conflicting schema
errors. Users can archive existing datasets and start new ones if old data is
important.

Previously additionalProperties API properties were RECORD fields. As these are
user defined unstructured data they would just keep adding to the BigQuery table
schema as more and more user defined fields were added over time. This change
converts any additionalProperties API property into an array of name value pairs
where the value is the type of the additionalProperties property. This should
greatly reduce the number of columns in BigQuery tables taken up by user defined
labels or app engine version names.

This is another fix for Issue #533

jaketf commented on tools/asset-inventory/README.md:
```diff
@@ -319,6 +319,75 @@ This repository contains some command line tools that let you run the export/imp
+
+### 1. Delete the dataset and recreate it.
+
+The simplist option, this will discard all your data losing prior history but will let you continue the import. You can also try deleting the tables that fail to import if just a few of them.
```
```suggestion
The simplest option, this will discard all your data losing prior history but will let you continue the import. You can also try deleting the tables that fail to import if just a few of them.
```

bmenasha replied:
done

bmenasha commented:
Thanks Jake.  I updated the version  and added notes in the README.md for how to deal with this issue with some example queries for this specific change.  Because of the nature of the change, i don't think it's easy to create a view, but it should be straightforward to change specific queries and I provided an example of that. 
thanks

jaketf commented on tools/asset-inventory/README.md:
```diff
@@ -319,6 +319,75 @@ This repository contains some command line tools that let you run the export/imp
+
+## Upgrading from version 1.0.0 to 2.0.0
+
+The 2.0.0 pipeline release unfortuantly changed the import schema to resovle [issue #533](https://github.com/GoogleCloudPlatform/professional-services/issues/533). Now some user specified and dynamic properties are represented as record arrays of key value pairs rather then just flat records. This was done to keep a more regular schema and prevent accumulation of columns from overflowing table li ...
```
```suggestion
The 2.0.0 pipeline release unfortuantely changed the import schema to resolve [issue #533](https://github.com/GoogleCloudPlatform/professional-services/issues/533). Now some user specified and dynamic properties are represented as record arrays of key value pairs rather then just flat records. This was done to keep a more regular schema and prevent accumulation of columns from overflowing table limits. This change could require changes in how you query the data for example, previously to query the App Engine traffic split across two versions you would write:
```

bmenasha replied:
great thanks. done with latest amend.
